### PR TITLE
[v6r17] Fixes in RMS/DMS + monitoring bucketting + Waiting site name

### PR DESCRIPTION
**Pull request by phicharp**
BEGINRELEASENOTES

* RMS
FIX: allow request to not fail if an SE is temporarily banned
CHANGE: remove scripts dirac-rms-show-request, dirac-rms-cancel-request and dirac-rms-reset-request and replace with a single script dirac-rms-request with option (default is "show")
CHANGE: allow script to finalize a request if needed and set the job status appropriately

* WMS
CHANGE: assign a job to Group.<site>,<country> if input files are at <site>.<country>. If several input replicas, assign Waiting to "MultipleInput"

* Configuration
FIX: allow default to be tuple, a dict or a set

* Monitoring
CHANGE: change bucket sizes and simplify setting

ENDRELEASENOTES

### CODE REVIEW

File: WorkloadManagementSystem/Executor/JobScheduling.py [hunk 1]
```diff
@@ -614,29 +616,16 @@ def __setJobSite( self, jobState, siteList ):
...
+    # If the job has input data, the online sites are hosting the data
+    if len( onlineSites ) == 1:
+      siteName = "Group.%s" % ".".join( onlineSites[0].split( "." )[1:] )
       self.jobLog.info( "Group %s is candidate" % siteName )
+    elif len( onlineSites ):
+      # More than one site with input
+      siteName = "MultipleInput"
+      self.jobLog.info( "Several input sites are candidate: %s" % ','.join( onlineSites ) )
     else:
+      # No input site reported (could be a user job)
```
Comment (fstagni) [inline WorkloadManagementSystem/Executor/JobScheduling.py]: So, anyway, we don't care anymore if the sites are Tier1 or not... correct?

File: DataManagementSystem/Agent/RequestOperations/RemoveFile.py [hunk 2]
```diff
@@ -79,20 +79,31 @@ def __call__( self ):
...
+    else:
+      bannedTargets = set()
 
     # # prepare waiting file dict
     # # We take only files that have no replica at the banned SEs... If no replica, don't
-    toRemoveDict = dict( ( opFile.LFN, opFile ) for opFile in waitingFiles if not bannedTargets.intersection( replicas.get( opFile.LFN, [] ) ) )
+    toRemoveDict = dict( ( opFile.LFN, opFile ) for opFile in waitingFiles \
+                         if not bannedTargets or not bannedTargets.intersection( replicas.get( opFile.LFN, [] ) ) )
+    # If some SEs are always banned, set Failed the files that cannot be removed
+    if bannedTargets and 'always banned' in self.operation.Error:
```
Comment (phicharp) [inline DataManagementSystem/Agent/RequestOperations/RemoveFile.py]: Well, we know what we have set as an error message ;-)

File: DataManagementSystem/Agent/RequestOperations/RemoveFile.py [hunk 3]
```diff
@@ -106,7 +117,7 @@ def __call__( self ):
         toRemoveDict = bulkRemoval["Value"]
 
       # # 2nd step - single file removal
-      for lfn, opFile in toRemoveDict.items():
+      for lfn, opFile in toRemoveDict.iteritems():
```
Comment (phicharp) [inline DataManagementSystem/Agent/RequestOperations/RemoveFile.py]: Done + a few others

File: WorkloadManagementSystem/Executor/JobScheduling.py (hunk 1 above)
Comment (phicharp) [inline WorkloadManagementSystem/Executor/JobScheduling.py]: exactly...

**Comment by andresailer** [comment]
For the release notes: just, e.g., "RMS" no "subsystem"

**Pull request merged**